hcipy/aperture/generic.py

```diff
@@ -60,12 +61,12 @@ def elliptical_aperture(diameters, center=None):
 		shift = center * np.ones(2)
 
 	def func(grid):
-		if grid.is_separated:
-			x, y = grid.separated_coords
-			f = ((((x - shift[0]) / (diameters[0] / 2))**2)[np.newaxis, :] + (((y - shift[1]) / (diameters[1] / 2))**2)[:, np.newaxis] <= 1).ravel()
-		else:
-			x, y = grid.as_('cartesian').coords
-			f = (((x - shift[0]) / (diameters[0] / 2))**2 + ((y - shift[1]) / (diameters[1] / 2))**2) <= 1
+
```

Comment:
No. The two different code paths in the original function were there for a reason. With separated grids you can get a HUGE boost in performance, as you don't work on the full nxn arrays, but only on the MUCH smaller n arrays. This means doing the division by a and b in the original linked equation _before_ adding x and y arrays together (which necessarily requires transitioning to the nxn arrays).

Please keep the two code paths for separated and non-separated grids.

Reply:
Also, no empty line here.

Reply:
I will have to add some extra code, because the separated grids can not be used for rotated ellipses. There are cross terms which are non separable. I think the rotated polygon issue has the same error.

Reply:
The equation is not completely separable, but you can keep it separable for as long as possible. Something like:

```python
x, y = grid.separated_coords
a, b = diameters / 2
h, k = shift

term1 = (((x - h) * cos(angle) / a))[np.newaxis, :] + ((y - k) * sin(angle) / a)[:, np.newaxis])**2
term2 = ...
f = (term1 + term2) <= 1
```

This is only 2+1=3 additions (two of which are broadcasted which are a little faster than non-broadcasted ones) and 2 multiplications of big arrays. Compare this to 3+3+1=7 additions and 2+2+2+2=8 multiplications of big arrays in your current code. So I'd expect the separated code to be ~3x faster. You can do the benchmarks to convince yourself of this. You can keep the current code for non-separated grids.

Also, please add a link to the webpage in the documentation. That equation is not trivial.

Reply:
This is not resolved. My original comment stands. You've now just added a separate version for angle=0, 90, 180 or 270 degrees. The other cases are just as slow as before. See my previous comment.

Reply:
I have sped up the code as much as I could. I ran a couple tests to check if the answers were still correct, and they all passed the tests. So I think this is done.

Reply:
Thank you! This is good now.